src/main/java/kitchenpos/menu/domain/MenuProducts.java

```diff
@@ -0,0 +1,36 @@
+package kitchenpos.menu.domain;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import javax.persistence.Transient;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+@Embeddable
+public class MenuProducts {
+
+    @OneToMany(mappedBy = "menu", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<MenuProduct> menuProducts = new ArrayList<>();
+
+    @Transient
+    private Price sum = Price.from(BigDecimal.ZERO);
+
+    protected MenuProducts() {}
+
+    public void add(final MenuProduct menuProduct) {
+        if (!menuProducts.contains(menuProduct)) {
+            this.menuProducts.add(menuProduct);
+        }
+        sum = sum.add(menuProduct.getProduct().getPrice(), menuProduct.getQuantity());
```

Comment:
menuProduct.getProduct().getPrice() [디미터 법칙](https://mangkyu.tistory.com/147) 위반이네요!
getter 를 사용하기 보다 다음과 같이 객체에 메세지를 보내보면 어떨까요? 😄 
```suggestion
        sum = sum.add(menuProduct.price());
```

Reply:
https://github.com/next-step/jwp-refactoring/pull/193#discussion_r667351953
수정하면서 해당 부분은 삭제되었습니다!

코멘트 주신 부분에 대해서는 menuProduct.productPrice() 와 같이 수정하였습니다!